src/Tuura/Boolean.hs

```diff
@@ -11,76 +11,73 @@ import Data.Foldable
 import Data.Maybe
 import Data.Ord
 
-type CNF a = [[Literal a]]
+newtype CNF a = CNF { fromCNF :: [[Literal a]] }
 
-type DNF a = [[Literal a]]
+newtype DNF a = DNF { fromDNF :: [[Literal a]] }
 
 data Literal a = Literal { variable :: a, polarity :: Bool } deriving (Eq, Ord)
 
-convertToCNF :: Eq a => (Expr a) -> CNF a
+convertToCNF :: Eq a => Expr a -> CNF a
 convertToCNF expr = cnf
   where
-    vars = toList expr
+    vars = nub $ toList expr
     values = mapM (const [False, True]) vars
-    fs = filter (\v -> not $ eval expr (\x -> getValue x vars v)) values
-    genVar val var = Literal var val
-    sim = simplifyCNF $ map (\v -> nub $ (map (\f -> (genVar (v !! f) (vars !! f))) [0..(length vars - 1)])) fs
-    cnf = map (map (\s -> Literal (variable s) (not $ polarity s))) sim
+    fs = filter (not . eval expr . getValue vars) values
+    sim = CNF $ map (\v -> nub $ map (\f -> Literal (vars !! f) (v !! f)) [0..(length vars - 1)]) fs
+    cnf = CNF $ map (map (\s -> Literal (variable s) (not $ polarity s))) (fromCNF sim)
+    getValue vs vals v = fromJust $ lookup v $ zip vs vals
 
-genConcepts :: Bool -> [Literal String] -> String
-genConcepts v e
-    | (length e) == 1 = causes ++ op ++ effect
-    | otherwise       = "[" ++ causes ++ "]" ++ op ++ effect
+-- Generates a concept for output signal to rise if v = True, fall if v = False
+genConcepts :: Bool -> String -> [Literal String] -> String
```

Comment:
Transition will make it clearer, and I have done this.
Instead of just doing `show Transition String`, I just use the value of the transition to determine 
`rise` or `fall`.